src/Core/Resolvers/QueryExecutor.cs

```diff
@@ -158,7 +162,8 @@ public class QueryExecutor<TConnection> : IQueryExecutor
             Func<DbDataReader, List<string>?, Task<TResult>>? dataReaderHandler,
             string dataSourceName,
             HttpContext? httpContext = null,
-            List<string>? args = null)
+            List<string>? args = null,
+            IMiddlewareContext? middlewareContext = null)
```

Comment:
You already have access to "HttpContext" can't you just add the required metadata to that? This pr adds another function argument in many places just to keep track of a timer seems excessive.